src/components/TokenTranslationUpdate/TokenTranslationUpdate.js

```diff
@@ -0,0 +1,275 @@
+        .then(function (response) {
+          if(response.data.translation){ 
+            setTokenTranslation(response.data.translation);
+          }else{
+            setTokenTranslation('');
+          }
+        })
+        .catch((error) => {
+          setResponseStatus([true, 'error', error.response.data.error]);
+        });
+      }
+  };
+
+  const previous = () => {
+    if (tokenSelected > 1) {
+      setTokenSelected(tokenSelected - 1);
+      setOccurance(props.tokenDetail.occurrences[tokenSelected - 2]);
+      const sentenceId = props.tokenDetail.occurrences[tokenSelected-2];
+      const offset1 = sentenceId.offset[0];
+      const offset2 = sentenceId.offset[1];
+      API.get(
+        `autographa/project/token-translations?project_id=${projectId}&token=${token}&sentence_id=${sentenceId.sentenceId}&offset=${offset1}&offset=${offset2}`)
+        .then(function (response) {
+          if(response.data.translation){ 
+            setTokenTranslation(response.data.translation);
+          }else{
+            setTokenTranslation('');
+          }
+        })
+        .catch((error) => {
+          setResponseStatus([true, 'error', error.response.data.error]);
+        });
+    }
+  };
+
+  useEffect(() => {
+    const sentenceId = props.tokenDetail.occurrences[0];
+    const offset1 = sentenceId.offset[0];
+    const offset2 = sentenceId.offset[1];
+    API.get(
+      `autographa/project/token-translations?project_id=${projectId}&token=${token}&sentence_id=${sentenceId.sentenceId}&offset=${offset1}&offset=${offset2}`)
+      .then(function (response) {
+        if(response.data.translation){ 
+          setTokenTranslation(response.data.translation);
+        }else{
+          setTokenTranslation('');
+        }
+      })
+      .catch((error) => {
+        setResponseStatus([true, 'error', error.response.data.error]);
+      });
+      setOccurance(sentenceId)
+  }, []);
+
+
+  const tokenApiCall = (data) =>{
+    API.put(`autographa/project/tokens?project_id=${projectId}`, data)
+      .then(function (response) {
+        setResponseStatus([true, 'success', response.data.message]);
+        if (response.data.message === 'Token translations saved') {
+          next();
+        }
+      })
+      .catch((error) => {
+        setResponseStatus([true, 'error', error.response.data.error]);
+      });
+  }
+
+  const tokenReplace = (replaceType) => {
+    if (replaceType === 'oneToken') {
+      const data = [
+        { token: token, occurrences: [occurance], translation: tokenTranslation },
+      ];
+      tokenApiCall(data)  
+    }else if(replaceType === 'replaceAll'){
+      const allOccurance = props.tokenDetail.occurrences;
+      const data = [
+        { token: token, occurrences:allOccurance , translation: tokenTranslation },
+      ];
+      tokenApiCall(data);
+      handleDialogClose();
+    }else if(replaceType === 'remainingTokens'){
+      const remOccurance = (props.tokenDetail.occurrences).slice(tokenSelected-1);
```

Comment:
remaining occurances needs to be calculated using all - already translated tokens